src/engine/Util/Detector.ts

```diff
@@ -1,13 +1,134 @@
 /// <reference path="Log.ts" />
+
+/**
+ * This is the list of features that will be used to log the supported
+ * features to the console when Detector.logBrowserFeatures() is called.
+ */
+const REPORTED_FEATURES: Object = {
+   webgl: 'WebGL',
+   webaudio: 'WebAudio',
+   gamepadapi: 'Gamepad API'
+};
+
 module ex {
 
    /**
+    * Interface for detected browser features matrix
+    */
+   export interface IDetectedFeatures {
+      canvas: boolean;
+      arraybuffer: boolean;
+      dataurl: boolean;
+      objecturl: boolean;
+      rgba: boolean;
+      webaudio: boolean;
+      webgl: boolean;
+      gamepadapi: boolean;
+   }
+
+   /**
     * Excalibur internal feature detection helper class
     */
    export class Detector {
-      
+
+      private _features: IDetectedFeatures = null;
+
       public failedTests: string[] = [];
+
+      public constructor() {
+         this._features = this._loadBrowserFeatures();
+      }
+
+      /**
+       * Returns a map of currently supported browser features. This method
+       * treats the features as a singleton and will only calculate feature
+       * support if it has not previously been done.
+       */
+      public getBrowserFeatures(): Object {
```

Comment:
Also, optionally, we could use a ES5 getter here since the result is cached once when the game starts and only returns the cached response from then on. I'm cool either way since this is mostly an internal function.